R/tm_a_regression.R

```diff
@@ -1037,25 +1033,26 @@ srv_a_regression <- function(id,
     )
 
     ### REPORTER
-    if (with_reporter) {
-      card_fun <- function(comment, label) {
-        card <- teal::report_card_template(
-          title = "Linear Regression Plot",
-          label = label,
-          with_filter = with_filter,
-          filter_panel_api = filter_panel_api
-        )
-        card$append_text("Plot", "header3")
-        card$append_plot(plot_r(), dim = pws$dim())
-        if (!comment == "") {
-          card$append_text("Comment", "header3")
-          card$append_text(comment)
-        }
-        card$append_src(source_code_r())
-        card
-      }
-      teal.reporter::simple_reporter_srv("simple_reporter", reporter = reporter, card_fun = card_fun)
-    }
+    card_fun <- reactive({
+      req(plot_r(), source_code_r())
+      teal.reporter::report_document(
+
+        "## Plot",
+        plot_r(),
+        "## Source Code",
+        format_as_markdown_chunk(source_code_r()),
```

Comment:
There is no such thing in markdown syntax as "chunk", this term is rmarkdown specific. So the word markdown and chunk can't go along. I suggest `as.code_chunk` or [`as.code_block`](https://www.markdownguide.org/basic-syntax/#code-blocks-1). This function should be in `teal.reporter` because this will be necessary in all teal.module.*

```suggestion
        teal.reporter::as.code_block(source_code_r()),
```